docs(INSTALL): Bring back autotools instructions

zetok opened the pull request:
Reverted, since apparently cmake is not supposed to be used for things
other than testing/development and causes client build failures.

Apparently making it work for clients would require complicating
maintenance, which clearly can't be done.

This reverts commit 48ddb115995f2bb55a736ae4cc54c70f3d34eeb4.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/toktok/c-toxcore/376)
<!-- Reviewable:end -->

zetok commented:
cc @Encrypt @nurupo could you revert toxcore to autotools on jenkins?

jin-eld commented:
As a side note: when you add cmake compiler options or #defines that are supposed to be controllable by the user, let me know or better, add a new issue and assign it to me. I will make sure to add those things to autotools so that both build systems stay in sync in terms of option configuration.

cebe commented:
> apparently cmake is not supposed to be used for things
> other than testing/development and causes client build failures.

@zetok can you give more details about the issues it caused? maybe they are solveable?

robinlinden commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[INSTALL.md, line 466 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/376#-K_UpfQltEJG9povk7Co:-K_UpfQmYiKgvzQjn-5J:bllra0a) ([raw file](https://github.com/toktok/c-toxcore/blob/d3da6daff7ef813289126f19ced0898d1071300b/INSTALL.md#L466)):*
> ```Markdown
> You can also use a precompiled win32 binary of libsodium, however you will have to place the files in places where they can be found, i.e., dll's go to /bin headers to /include and libraries to /lib directories in your MinGW shell.
> 
> Next, install toxcore library, should either clone this repo by using git, or just download a [zip of current Master branch](https://github.com/irungentoo/toxcore/archive/master.zip) and extract it somewhere.
> ```

Why would we tell people to download irungentoo's toxcore on the TokTok repo?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/376)*
<!-- Sent from Reviewable.io -->

zetok commented:
@cebe there were relevant logs on jenkins, now they're probably gone.

According to @iphydf it's not possible to have a maintainable toxcore and usable cmake build at the same time.

I don't really know, I just updated docs to reflect that.

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[INSTALL.md, line 466 at r1](https://reviewable.io:443/reviews/toktok/c-toxcore/376#-K_UpfQltEJG9povk7Co:-K_UrDYwanzsTyYuufpe:bu52oq3) ([raw file](https://github.com/toktok/c-toxcore/blob/d3da6daff7ef813289126f19ced0898d1071300b/INSTALL.md#L466)):*
<details><summary><i>Previously, robinlinden (Robin Lindén) wrote…</i></summary><blockquote>

Why would we tell people to download irungentoo's toxcore on the TokTok repo?
</blockquote></details>

Whops, revert gone wild. Fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/376#-:-K_UqhYGHsrK1tuL1VuV:b-xgdiin)*
<!-- Sent from Reviewable.io -->

nurupo commented:
>cc @Encrypt @nurupo could you revert toxcore to autotools on jenkins?

I could, it would be just a huge pain since it would break Toxic and uTox builds, as they use cmake toxcore, so it would require coordination from @JFreegman and @GrayHatter.

So, before switching Jenkins to autotools, explain what exactly doesn't work in cmake and why shouldn't we just fix it and continue use cmake on Jenkins?

zetok commented:
@nurupo since you seem to have forgotten what doesn't work: https://github.com/qTox/qtox-irc-logs/blob/4dcc37baf6eea28b5ca19ba5be86243b359b630e/2016/12/%23qtox_20161221.log.txt#L46

@tux3 hit the same issue yesterday.

>So, before switching Jenkins to autotools, explain what exactly doesn't work in cmake and why shouldn't we just fix it and continue use cmake on Jenkins?

Clearly, linking doesn't work. As for why it shouldn't be fixed to work just like autotools do, perhaps @iphydf will answer that.

nurupo commented:
>Clearly, linking doesn't work. As for why it shouldn't be fixed to work just like autotools do, perhaps @iphydf will answer that.

It's linking in qTox which doesn't work though. @tux3 has fixed it in https://github.com/qTox/qTox/pull/4038. I suggest closing this issue as it's not an issue in toxcore but in qTox.

zetok commented:
@nurupo You've missed the point that cmake toxcore breaks stuff that works with autotools. This is purely a toxcore issue.

GrayHatter commented:
Works in uTox. This is a qTox issue.

zetok commented:
@GrayHatter 
>Works in uTox. This is a qTox issue.

As I've mentioned above, it's a toxcore issue.

If you're insisting though that it should be closed since it works for some people while it doesn't for other, why not close your own PR (#368)? After all, ToxMe works not just for some people, but for everyone except µTox. Makes a lot more sense to close it.

Reopening. Feel free to close it right after closing your own PR, or if there's a consensus that making ~tox development harder with cmake is must, and easier development with autotools is not allowed.

iphydf commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> please rebase. Also, could you create an issue for updating these docs for cmake? We can't fully move to cmake before all important use cases (in particular, cross compilation) are covered.

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/376#-:-K_ZqavPqD4UXWM90eSb:bh4gz9y)*
<!-- Sent from Reviewable.io -->

zetok commented:
Rebased.

cebe commented:
Still would be good to have a description of the issue you face with cmake to be able to solve it.

iphydf commented:
@nurupo is working on proper cmake instructions. Would you prefer we avoid mentioning it until those instructions are done?

zetok commented:
>Would you prefer we avoid mentioning it until those instructions are done?

Documentation of functionality is always a good thing. Cmake usage should be documented, and IMO it shouldn't be delayed until $certain_point if the functionality is already present.

Cmake docs probably should have a remark somewhere that it might not be yet providing the same level of functionality for client devs as autotools.

Regarding making an issue, I'll do it once this PR gets merged (wouldn't make much sense to make it right now since docs haven't been affected yet).

iphydf commented:
<img class="emoji" title=":lgtm_strong:" alt=":lgtm_strong:" align="absmiddle" src="https://reviewable.io/lgtm_strong.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/toktok/c-toxcore/376#-:-K_hxtXcwah4W-wnkAew:b-jbsf8z)*
<!-- Sent from Reviewable.io -->
